lib/project_types/extension/extension_project_keys.rb

```diff
@@ -4,7 +4,7 @@
 module Extension
   module ExtensionProjectKeys
     REGISTRATION_ID_KEY = "EXTENSION_ID"
-    EXTENSION_TYPE_KEY = "EXTENSION_TYPE"
```

Comment:
Externally we're still using the term "extension type" to avoid breaking existing projects and because without Shopify context the term is still more self explanatory than "specification identifier".